modules/contact/src/userobjects/PenaltyFrictionUserObject.C

```diff
@@ -0,0 +1,353 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PenaltyFrictionUserObject.h"
+#include "MooseVariableFE.h"
+#include "SystemBase.h"
+#include "MortarUtils.h"
+#include "MooseUtils.h"
+#include "MortarContactUtils.h"
+
+registerMooseObject("ContactApp", PenaltyFrictionUserObject);
+
+InputParameters
+PenaltyFrictionUserObject::validParams()
+{
+  InputParameters params = WeightedVelocitiesUserObject::validParams();
+  params.addClassDescription(
+      "Computes the mortar frictional contact force via a penalty approach.");
+  params.addRequiredParam<Real>("penalty", "The penalty factor for normal interaction.");
+  params.addParam<Real>("penalty_friction",
+                        0.0,
+                        "The penalty factor for frictional interaction. If not provide, the normal "
+                        "penalty factor is also used for the frictional problem.");
+  params.addRequiredParam<Real>("friction_coefficient",
+                                "The friction coefficient ruling Coulomb friction equations.");
+  return params;
+}
+
+PenaltyFrictionUserObject::PenaltyFrictionUserObject(const InputParameters & parameters)
+  : WeightedVelocitiesUserObject(parameters),
+    _penalty(getParam<Real>("penalty")),
+    _penalty_friction(isParamSetByUser("penalty_friction") ? getParam<Real>("penalty_friction")
```

Comment:
```suggestion
    _penalty_friction(isParamValid("penalty_friction") ? getParam<Real>("penalty_friction")
```

Reply:
The parameter is always going to be valid, isn't it? I thought a default value would make it valid. I am interested to check if the user made the effort to set it in the input file.

Reply:
I see this comment is linked to the next one.